docs/DeveloperGuide.md

```diff
@@ -1,38 +1,361 @@
+
+The `Transactions` component
+
+- A `Transaction` is recorded upon every command that changes the balance of an account
+- Certain commands allow the provision of a description parameter to associate a custom description to a transaction
+- `Transactions` are only accessed through the `TransactionManager`
+- `Transactions` associated to an account are deleted if and only if the account is deleted
+
+## Implementation
+
+### Create/Delete account feature
+
+This feature is facilitated by `AccountList` Class within the `Accounts` Component.
+The method called from `AccountList` is the `addAccount` method which creates a new `Account` object.The `deleteAccount`
+method called from `AccountList` delete the specified `Account` object.
+
+The current implementation initialises the `Account` with 0 balance.
+Only currency account which 0 balance can be deleted.
+
+Given below is an example of the usage of this feature and the mechanism at each step
+
+Step 1: The user launches the application for the first time and `AccountList` is created with no `Account`
+![AccountListObjectDiagram1](images/AccountListObjectDiagram1.png)
+
+Step 2: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is a valid string representing one
+of the elements of the `Currency` enum
+![AccountListObjectDiagram1](images/AccountListObjectDiagram2.png)
+
+Step 3: The user passes in the command `create-account <CURRENCY>`, where `CURRENCY` is also valid but different to that
+in step 1.
+![AccountListObjectDiagram1](images/AccountListObjectDiagram3.png)
+
+Step 4: The user passes the command `delete-account CURRENCY`, for example `delete-account SGD`.
+
+![AccountListObjectDiagram4](images/AccountListObjectDiagram4.png)
+
+The following sequence diagram shows how the Create Account operation works
+![AccountListObjectDiagram1](images/CreateAccountSeqDiagram.png)
+
+### Delete-account feature
+
+### Add/Withdraw money feature
+
+The add money(deposit) and withdraw money feature is facilitated by `AddCommand` and `WithdrawCommand` which both
+extends the `Command` class. With the provided input from user (`CURRENCY` and `AMOUNT`), `AddCommand`
+and `WithdrawCommand`
+update the balance of respective currency account accordingly.
+
+Step 1. The newly created `SGD` account has an initial balance of 0
+![AddWithdrawCommandObjectDiagram1](images/AddWithdrawCommandObjectDiagram1.png)
+
+Step 2. The user passes command `add CURRENCY AMOUNT` (eg. `add SGD 100`), where `CURRENCY` must be one of the available
+currency and `AMOUNT` must be positive numbers.
+![AddWithdrawCommandObjectDiagram2](images/AddWithdrawCommandObjectDiagram2.png)
+
+Step 3. The user passes command `withdraw` (eg. `withdraw SGD 25`), where `AMOUNT` must be smaller than the currency
+account balance.
+![AddWithdrawCommandObjectDiagram3](images/AddWithdrawCommandObjectDiagram3.png)
+
+The following sequence diagram shows how the add money operation works.
```

Comment:
Should :AddCommand be activated at the same time when execute() function calls it?

![image](https://user-images.githubusercontent.com/16986946/228763600-9791b7a1-41fb-4612-8903-77493eabe7d0.png)